monascaclient/keystone.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Hewlett Packard Enterprise Development LP
+//
+//    Licensed under the Apache License, Version 2.0 (the "License"); you may
+//    not use this file except in compliance with the License. You may obtain
+//    a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+package monascaclient
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"net/http"
+)
+
+type KeystoneConfig gophercloud.AuthOptions
+
+func SetKeystoneToken() error {
+	return monClient.setKeystoneToken()
+}
+
+func (c *Client) setKeystoneToken() error {
```

Comment:
Personally, I would rather this just returned the token and the code in client.go did things like headers.Add(). Seems strange to call this and have it overwrite the headers in the client